pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -200,13 +247,28 @@ func (r *ReconcileNodeMaintenance) Reconcile(request reconcile.Request) (reconci
 		return r.reconcileAndError(instance, err)
 	}
 
+	if r.isLeaseSupported {
+		reqLogger.Info("Lease object supported, obtaining lease")
+
+		proceed, reconcileRes := r.obtainLease(node, LeaseDurationInSeconds, reqLogger)
+		if !proceed {
```

Comment:
what happens when proceed == false ?  is that a valid situation ?  reconciliation will not occur again in this case.  could we have error scenarios here ?

Reply:
failing to obtain the lease is an error situation. Error is logged by obtainLease function if it occurs.